### PR TITLE
Add custom error message for legacy devices

### PR DESCRIPTION
**Pull request by kararade**
…s error in older browsers

**Release Type:** *Bug Fix* 

Fixes https://x-team-internal.atlassian.net/browse/XP-3643

## Description

It suggests the user use the latest browser version for errors caused in the following older browsers when they do not support `RequestAPI` and `URLSearchParams`:
- Chrome v48-
- Safari v9.1-
- Firefox v38-

## Checklist

**Before submitting a pull request,** please make sure the following is done:

- [x] set yourself as an assignee
- [x] set appropriate labels for a PR (`In Review` or `In Progress` depending on its status)
- [x] move respective JIRA issue to the `PEER REVIEW` column
- [x] make sure your code lints (`npm run lint`)
- [x] Flow typechecks passed (`npm run flow`)
- [x] Snapshots tests passed (`npm run jest`) (**NOTE:** This is temporarily disabled. Skip it.)
- [x] check cleanup tasks (https://github.com/x-team/xp/labels/cleanup) and take a suitable small one (if exists) in a related area of the current changes
- [ ] if added a new UI component that is meant to be used directly by other apps, remember to export that from the appropriate bundle file: `src/index.js` and/or `src/registration.js`
- [x] component's documentation (`.stories.js` file) is changed or added accordingly to reflect any new or updated use cases and/or debug cases
- [x] if you've fixed a bug, make sure to also include a new story that will expose declarations with problematic data that caused the bug in the first place, so that we can assure that no regressions will pop up on future interactions
- [x] if any snapshots have been changed, verify that component still works and looks as expected and update the changed snapshot (**NOTE:** This is temporarily disabled. Skip it.)
- [x] once finished with all the changes, you run `npm run build` command and committed changed files to the branch
- [x] **manually tested the app** by running it in several different browsers (Firefox, Chrome, Opera, Safari, MS IE/Edge, etc.) and checked nothing is broken and operates as expected!

## Steps to Test or Reproduce

In `xp-ui`:
1. Run `rm -r node_modules && npm i`
2. Run `npm run start` and open `http://localhost:9001/`
3. Scroll to the section "Core Components" > "ErrorBox" > "Debug" > "Custom error message for legacy devices"
4. Confirm the error displayed is "Please update your browser to the latest version" for the following points:
    - 'error \'TypeError: invalid \'instanceof\' operand y.Request\''
    - 'error \'TypeError: Expecting a function in instanceof check, but got undefined\''
    - 'error \'TypeError: y.Request is not a function. (evaluating \'t instanceof y.Request\')\''
    - 'error \'ReferenceError: URLSearchParams is not defined\''

## Impacted Areas in Application

- `ErrorBox.js`

## Screenshots

- 'error \'TypeError: invalid \'instanceof\' operand y.Request\''

Screen Shot on 2020-04-10 at 15:30:36

- 'error \'TypeError: Expecting a function in instanceof check, but got undefined\''

Screen Shot on 2020-04-10 at 15:30:49

- 'error \'TypeError: y.Request is not a function. (evaluating \'t instanceof y.Request\')\''

Screen Shot on 2020-04-10 at 15:30:59

- 'error \'ReferenceError: URLSearchParams is not defined\''

Screen Shot on 2020-04-10 at 15:31:09

### CODE REVIEW

File: src/components/ui/ErrorBox.js [hunk 1]
```diff
@@ -63,6 +64,10 @@ class ErrorBox extends PureComponent<Props> {
     const keys = Object.keys(errors)
     if (!keys.length) { return <div /> }
 
+    if (errors && errors.name && isOlderBrowserError(errors.name)) {
```
Comment (nicksp) [inline src/components/ui/ErrorBox.js:67]: How do we know that the error we discussed will be passed as a `name` prop? Have you verified that on devices after added the change using BrowserStack or anything like that? The current problem is that `error` might have random structure (as you can see in `xp-registration` or `xp` projects) here and we should ensure the message we're dealing with is always passed in a `name` prop. Is there a way to test thins locally?

Comment (kararade) [reply src/components/ui/ErrorBox.js]: hey @nicksp 

I have updated the code to dynamically obfuscate these identified errors in any key.

Cheers!

**Comment by kararade** [comment]
> The only thing that is left before we can merge, is verifying we tested the app (not the storybook story) on devices you used initially via BrowserStack to verify it behaves as expected.

hey @nicksp 

Here some screenshots using different browsers with BrowserStack

**MAC**

- CHROME v41

Mac_chrome41

- CHROME v48

Mac_chrome48

- FIREFOX v38

Mac_firefox38

- SAFARI v9.1 

Mac_safari9 1

**WINDOWS**

- CHROME v41

Windows_chrome41

- CHROME v48
Windows_chrome48

- FIREFOX v38

Windows_firefox38

**IOS 9**

- SAFARI

iphone_os9_safari

- CHROME

iphone_os9_chrome

- FIREFOX

iphone_os9_firefox

**IOS 8**

- SAFARI

iphone_os8_safari

- CHROME

iphone_os8_chrome

Thanks!

**Comment by nicksp** [comment]
> Here some screenshots using different browsers with BrowserStack

@kararade Fantastic, those screenshots were super helpful and insightful 👏 Thanks!

**Pull request merged**